### ISSUE TITLE
complete backend rewrite/refactor

### ISSUE THREAD

**Issue by oohnoitz** [oohnoitz/jii#17]
The goal of this ticket was to do a complete refactor/rewrite of jii. It's to trim down the amount of dependencies required and make it a bit more maintainable.

**Tasks**
- [x] remove Hapi.js framework => replace with Express.js
- [x] update engine requirement to Node.js 6.0+
- [x] update to use ES6/ES7 features
- [ ] use MongoDB driver instead of Mongoose
- [ ] add tests

**TODO**
- [ ] re-implement link mirroring
- [ ] standardize error/response and api

**Comment by BlueAccords** [comment]
Just a question but what alternatives are are you planning on using instead of Hapi.js?

**Comment by oohnoitz** [comment]
I haven't taken the time to actually look into any other alternatives yet. There is a possibility of just using the HTTP/HTTPS modules with a package that handles requests/multi-part. I'm open to suggestions.

**Comment by BlueAccords** [comment]
Well what's the current problem with using hapi.js?

**Comment by oohnoitz** [comment]
It isn't an issue with the framework but mostly lack of focus/time I have for this project. There are a few more important aspects of the code I would like to refactor/rewrite instead though. I simply wrote it as a whim back when I made this issue.

Previously, I had to spent hours checking the change logs and making sure nothing broke while upgrading the framework to the latest version. While this is practically normal when upgrading to the latest major version, I wanted to try and minimize the amount of time doing such work on this project. There's also the lack of tests and some awkward design decisions in this project (again not an issue with the framework) that ended up making it a painful experience the few times I've done it, but that's on me.
